CHANGELOG.md

```diff
@@ -18,6 +18,7 @@ IMPROVEMENTS:
   * drivers: Interpret Nomad variables in environment variables/args [GH-653]
   * core: Populate job status [GH-663]
   * core/cli: Print short identifiers and UX cleanup [GH-675, GH-693, GH-692]
+  * api: Allow users to set arbitrary headers via agent config [GH-699]
```

Comment:
Don't we normally handle the CHANGELOG outside of the PR itself? I think this should also be `core/api` as we have used before.